File: src/main/java/org/app/infrastructure/endpoints/dto/GetUserItemDTO.java
package org.app.infrastructure.endpoints.dto;

import org.app.domain.User;

public class GetUserItemDTO {
    public String id;
    public String username;

    public GetUserItemDTO(String id, String username) {
        this.id = id;
        this.username = username;
    }

    public GetUserItemDTO(User user) {
        this.id = user.getId().toString();
        this.username = user.getUsername();
    }
}

File: go/app/main/infrastructure/endpoints/dto/get_user_item_dto.go
package dto

import "mutualAttentionSystem/app/main/domain"

type GetUserItemDTO struct {
    Id       string `json:"id"`
    Username string `json:"username"`
}

func ToGetUserItem(user *domain.User) *GetUserItemDTO {
    return &GetUserItemDTO{user.ID.String(), user.Username}
}
